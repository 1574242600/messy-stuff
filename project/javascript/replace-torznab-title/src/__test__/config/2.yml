rules: 
  name:
    url: https://example.com/
    repRule:
      repRule1:
        is: match
        parse: '[(.*?)][(.*?)]'
      repRule2:
        is: match
        parse: '(.*?)\(.*?)'
        template: '[$1] [$2]'