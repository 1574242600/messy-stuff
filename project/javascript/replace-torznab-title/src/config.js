import * as fs from "fs";
import yaml from "js-yaml";
import { z } from "zod";

const { record, object, number, string } = z;
const configSchema = object({
    port: number().default(7724),
    host: string().default("127.0.0.1"),
    rules: record(
        object({
            url: string().url(),
            repRule: record(
                object({
                    is: string(),
                    parse: string(),
                    template: string().optional(),
                })
            ),
        })
    ),
});

let config;

function loadConfig(path) {
    try {
        const str = fs.readFileSync(path, "utf8");
        config = yaml.load(str);
        config = configSchema.parse(config);
    } catch (e) {
        throw new Error(`Failed to load config file: ${e.message}`);
    }
}

export { config, loadConfig };