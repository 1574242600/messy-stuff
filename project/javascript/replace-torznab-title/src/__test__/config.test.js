import { loadConfig, config } from "../config";

test("loadConfig", () => {
    loadConfig("src/__test__/config/1.yml");
    
    expect(config).toMatchSnapshot();
});