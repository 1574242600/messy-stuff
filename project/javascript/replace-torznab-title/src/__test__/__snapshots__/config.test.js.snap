// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadConfig 1`] = `
Object {
  "host": "127.0.0.1",
  "port": 7724,
  "rules": Object {
    "name": Object {
      "repRule": Object {
        "repRule1": Object {
          "is": "match",
          "parse": "[(.*?)][(.*?)]",
        },
        "repRule2": Object {
          "is": "match",
          "parse": "(.*?)\\\\(.*?)",
          "template": "[$1] [$2]",
        },
      },
      "url": "https://example.com/",
    },
  },
}
`;
